Common/Brokerages/DefaultBrokerageModel.cs

```diff
@@ -236,5 +242,22 @@ public virtual ISettlementModel GetSettlementModel(Security security, AccountTyp
             return new ImmediateSettlementModel();
         }
 
+        protected bool ValidateQuantityFloored(decimal quantity, out BrokerageMessageEvent message)
+        {
+            message = null;
+
+            // validate quantity floored
+            if (quantity != Math.Floor(quantity))
```

Comment:
Cool stuff! We might need ceiling for shorts (negative quantities), but since we're not checking buying power here maybe its if (quantity != floor && quantity != ceiling) to confirm that its a whole number. 

I think we can express that as "Math.Abs(quantity % 1) > 0"